source/compiler-core/slang-nvrtc-compiler.cpp

```diff
@@ -228,24 +228,68 @@ static SlangResult _parseNVRTCLine(SliceAllocator& allocator, const UnownedStrin
     if (split.getCount() >= 3)
     {
         // tests/cuda/cuda-compile.cu(7): warning: variable "c" is used before its value is set
-
         const auto split1 = split[1].trim();
 
-        if (split1 == "error")
+        Severity severity = Severity::Unknown;
+
+        if (split1 == toSlice("error") || 
+            split1 == toSlice("catastrophic error"))
+        {
+            severity = Severity::Error;
+        }
+        else if (split1 == toSlice("warning"))
+        {
+            severity = Severity::Warning;
+        }
+        else if (split1.indexOf(toSlice("error")) >= 0)
```

Comment:
might be better to use "error:" since to avoid triggering this logic in situations like `warning: variable "errorCode" is not used`.

Reply:
Fair point, I'll add a more constrained test.